File: javascript/coding-challenges/cc-006/solution/romanNumeralsSolution.js
const roman = {
  M: 1000,
  CM: 900,
  D: 500,
  CD: 400,
  C: 100,
  XC: 90,
  L: 50,
  XL: 40,
  X: 10,
  IX: 9,
  V: 5,
  IV: 4,
  I: 1,
};
function toRoman(n) {
  return Object.keys(roman)
    .map((key) => {
      let res = "";
      while (n >= roman[key]) {
        res += key;
        n -= roman[key];
      }
      return res;
    })
    .join("");
}

console.log(toRoman(457));

File: javascript/coding-challenges/cc-006/solution/romanNumeralsSolution.py
configuration = {
	1000: "M",
	900: "CM",
	500: "D",
	400: "CD",
	100: "C",
	90: "XC",
	50: "L",
	40: "XL",
	10: "X",
	9: "IX",
	5: "V",
	4: "IV",
	1: "I"
}

def numeral(number):
	roman = []
	for size in sorted(configuration,reverse=True):
		qty = int(number / size)
		number = number % size
		roman.append(configuration[size] * qty)
	return "".join(roman)

print(numeral(148))